backend/server/src/db/service.rs

```diff
@@ -9,7 +9,9 @@ use super::model::{build::DrvBuildMetadata, ForInsert};
 
 #[derive(Clone)]
 pub struct DbService {
-    pool: SqlitePool,
+    // Instead of exposing this, we should probably have a function
+    // where people can get a cloned instance
+    pub pool: SqlitePool,
```

Comment:
I think this should not be publicly accessible at all. Imo the codebase will be easier maintainable if all accesses of the db need to go through this type.

I think you need it public because you added an impl in another module? Maybe it makes sense to use the repository pattern instead (i.e. have one type for each data domain that implements db functions for this domain). The db service could then just bundle all the repositories.

Reply:
made pure fuctions in model::drv, and added methods which pass a pool to them.

Reply:
> what would that look like?

Let's keep it like this for now and I will show you what I meant in my upcoming PR for #33, which will also add some db accessing functions.